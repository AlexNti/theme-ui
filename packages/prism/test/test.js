import React from 'react'
import { render } from '@testing-library/react'
import Prism from '../src'

const CODE = `
  console.log('hello, world!')
`

test('renders a code block', () => {
  const result = render(<Prism className="language-js">{CODE}</Prism>)

  expect(result).toMatchSnapshot()
})

test('renders with other languages', () => {
  const json = render(
    <Prism
      className='language-php'
      children='<h1>Hello</h1>'
    />
  )
  console.log(json)
  expect(json).toMatchSnapshot()
})
