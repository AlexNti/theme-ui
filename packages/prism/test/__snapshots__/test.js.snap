// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a code block 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <pre
        class="prism-code language-js css-0"
      >
        <div
          class="token-line css-0"
        >
          <span
            class="token console class-name css-0"
          >
            console
          </span>
          <span
            class="token punctuation css-0"
          >
            .
          </span>
          <span
            class="token method function property-access css-0"
          >
            log
          </span>
          <span
            class="token punctuation css-0"
          >
            (
          </span>
          <span
            class="token string css-0"
          >
            'hello, world!'
          </span>
          <span
            class="token punctuation css-0"
          >
            )
          </span>
        </div>
      </pre>
    </div>
  </body>,
  "container": <div>
    <pre
      class="prism-code language-js css-0"
    >
      <div
        class="token-line css-0"
      >
        <span
          class="token console class-name css-0"
        >
          console
        </span>
        <span
          class="token punctuation css-0"
        >
          .
        </span>
        <span
          class="token method function property-access css-0"
        >
          log
        </span>
        <span
          class="token punctuation css-0"
        >
          (
        </span>
        <span
          class="token string css-0"
        >
          'hello, world!'
        </span>
        <span
          class="token punctuation css-0"
        >
          )
        </span>
      </div>
    </pre>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
